File: android-java-foundation/src/main/java/com/frontend/state/FiniteStateEntity.java
// ======================================================================
// Project Name    : android_foundation
//
// Copyright © 2018 U-CREATES. All rights reserved.
//
// This source code is the property of U-CREATES.
// If such findings are accepted at any time.
// We hope the tips and helpful in developing.
// ======================================================================
package com.frontend.state;
public class FiniteStateEntity<T> {
    public boolean isNewState;
    public FiniteState<T> state;
    public String currentStateName;
    public String previousStateName;
    public FiniteStateEntity() {
        this.isNewState = false;
        this.state = null;
        this.currentStateName = "";
        this.previousStateName = "";
    }
    public void update(String stateName, FiniteState<T> nextState) {
        if (this.currentStateName != stateName) {
            this.previousStateName = this.currentStateName;
            this.currentStateName = stateName;
            this.state = nextState;
        }
        this.isNewState = true;
        return;
    }
}

File: android-kotlin-foundation/src/main/java/com/frontend/state/FiniteStateEntity.kt
// ======================================================================
// Project Name    : android_foundation
//
// Copyright © 2018 U-CREATES. All rights reserved.
//
// This source code is the property of U-CREATES.
// If such findings are accepted at any time.
// We hope the tips and helpful in developing.
// ======================================================================
package com.frontend.state
open class FiniteStateEntity<T> {
    public var isNewState: Boolean = false
        get
        set
    public var state: FiniteState<T>? = null
        get
        set
    public var currentStateName: String = ""
        get
        set
    public var previousStateName: String = ""
        get
        set
    open fun update(stateName: String, nextState: FiniteState<T>) {
        if (this.currentStateName != stateName) {
            this.previousStateName = this.currentStateName
            this.currentStateName = stateName
            this.state = nextState
        }
        this.isNewState = true
        return
    }
}